woocommerce/pinterest-for-woocommerce
# Retry mechanism for feed generation

### Changes proposed in this Pull Request:

Closes #563.

An error during the feed generation breaks the generation process. This PR adds a retry mechanism to avoid that the generation process breaks.

### Screenshots
![image](https://user-images.githubusercontent.com/40774170/211824059-789bbd88-5f22-4861-a28b-bd67bc9fcfb9.png)


### Detailed test instructions:
1. Install a build version of this PR.
2. Add the following snippet to the site (the snippet will break the feed generation in the first batch):
```
add_filter( 'pinterest_for_woocommerce_excluded_product_types', function() {
	throw new Exception( 'This is a test Exception' );
} );
```
3. The feed generation should apply the retry mechanism on the first batch after the failure.
4. The retry details should be available in the logs (`pinterest-for-woocommerce-sync`).


### Changelog entry

> Fix - Add retry mechanism for the feed generation.
